### PR TITLE
[format_specifier] - Fixes ESP_LOG format specifier mismatches in components/esp_matter (CON-414)

### PR DESCRIPTION
**Pull request by dhairyashah1**
Aim - To resolve format specifier mismatches in `ESP_LOG` prints for `endpoint_id`, `cluster_id`, `attribute_id` & `command_id`. Solves #296 

Change Overview - 
- Fixes in `components/esp_matter` files - [`esp_matter_attribute_utils.cpp`](https://github.com/espressif/esp-matter/blob/main/components/esp_matter/esp_matter_attribute_utils.cpp), [`esp_matter_core.cpp`](https://github.com/espressif/esp-matter/blob/main/components/esp_matter/esp_matter_core.cpp) & [`esp_matter_command.cpp`](https://github.com/espressif/esp-matter/blob/main/components/esp_matter/esp_matter_command.cpp) 
- Fixes in `examples/zigbee_bridge/main` - [`zigbee_bridge.cpp`](https://github.com/espressif/esp-matter/blob/main/examples/zigbee_bridge/main/zigbee_bridge.cpp)

Tests - 
- Sucessfully compiled & commissioned [`light_example`](https://github.com/espressif/esp-matter/tree/main/examples/light) on `esp32s3` with and without the following compile option in [`CMakeLists.txt`](https://github.com/espressif/esp-matter/blob/main/examples/light/CMakeLists.txt#L54)
```
idf_build_set_property(COMPILE_OPTIONS "-Wno-format-nonliteral;-Wno-format-security;-Wformat=0" APPEND)
```

### CODE REVIEW

File: components/esp_matter/esp_matter_core.cpp [hunk 1]
```diff
@@ -1306,14 +1306,14 @@ esp_err_t store_val_in_nvs(attribute_t *attribute)
     char nvs_namespace[16] = {0};
     char attribute_key[16] = {0};
     snprintf(nvs_namespace, 16, "endpoint_%X", endpoint_id); /* endpoint_id */
-    snprintf(attribute_key, 16, "%X:%X", cluster_id, attribute_id); /* cluster_id:attribute_id */
+    snprintf(attribute_key, 16, "%lX:%lX", cluster_id, attribute_id); /* cluster_id:attribute_id */
```
Comment (shubhamdp) [inline components/esp_matter/esp_matter_core.cpp:1309]:
```suggestion
    snprintf(attribute_key, 16, "%" PRIx32 ":%" PRIx32, cluster_id, attribute_id); /* cluster_id:attribute_id */
```

File: components/esp_matter/esp_matter_core.cpp [hunk 2]
```diff
@@ -1348,14 +1348,14 @@ esp_err_t get_val_from_nvs(attribute_t *attribute, esp_matter_attr_val_t *val)
...
```
Comment (shubhamdp) [inline components/esp_matter/esp_matter_core.cpp:1351]:
```suggestion
    snprintf(attribute_key, 16, "%" PRIx32 ":%" PRIx32, cluster_id, attribute_id); /* cluster_id:attribute_id */
```

**Review by shubhamdp — APPROVED** [review]
Minor comments, LGTM!

File: components/esp_matter/esp_matter_core.cpp (hunk 2 above)
Comment (dhairyashah1) [reply components/esp_matter/esp_matter_core.cpp]: Made suggested changes :+1:

**Pull request merged**